TTYC.Application/Users/Queries/ValidatePassword/ValidatePasswordHandler.cs

```diff
@@ -0,0 +1,24 @@
+﻿using MediatR;
+using TTYC.Persistence;
+
+namespace TTYC.Application.Users.Queries.ValidatePassword
+{
+	public class ValidatePasswordHandler : IRequestHandler<ValidatePasswordQuery, bool>
+	{
+		private readonly ApplicationDbContext dbContext;
+
+		public ValidatePasswordHandler(ApplicationDbContext dbContext)
+		{
+			this.dbContext = dbContext;
+		}
+
+		public Task<bool> Handle(ValidatePasswordQuery query, CancellationToken cancellationToken)
+		{
+			var user = dbContext.Users.FirstOrDefault(x => x.PhoneNumber == query.UserName);
+
```

Comment:
lets discuss this. I am not sure this gives us some advantage comparing to async await usage